# Estimaciones para la Solucion Automatica y Semi-Automatica.

In [164]:
import pandas as pd
import numpy as np
import math

## Estimacion de Proporciones:

In [165]:
tBase = 118769

In [166]:
# Proporcion de Imagenes con Animales
pAnimal = 0.3188
print('Proporcion de Imagenes con Animales: ', round(pAnimal * 100,2), '%')

Proporcion de Imagenes con Animales:  31.88 %


In [167]:
# Proporcion de Imagenes sin Animales
pNoAnimal = 1 - pAnimal
print('Proporcion de Imagenes sin Animales: ', round(pNoAnimal * 100,2), '%')

Proporcion de Imagenes sin Animales:  68.12 %


In [168]:
pGuanaco = (49.19/100)

# Solucion Automatica:

## Tiempo Total del Proceso:

In [169]:
# 6480 imagenes por hora
tAutom = 60*60/6480
TAutom = (tBase * (tAutom))/(60*60)
print('Tiempo de Automatizacion: ', round(tAutom,2), 'segundos por imagen')
print('Tiempo de Automatizacion: ', math.ceil(TAutom/8), 'jornadas')

Tiempo de Automatizacion:  0.56 segundos por imagen
Tiempo de Automatizacion:  3 jornadas


### Cantidad de Imagenes por Hora:

6480 Imagenes por Hora segun Estudios

In [170]:
print('Tiempo de Automatizacion: ', round(tAutom,2), 'segundos')

cHora = 60*60/(tAutom)
print('Cantidad de Imagenes por Hora: ', math.ceil(cHora))

Tiempo de Automatizacion:  0.56 segundos
Cantidad de Imagenes por Hora:  6480


### Precision de Etiquetado:

Segun Estudios:
- Error FP: 2.49%
- Error FN: 3.8%
- Error Especies: 5.1%
- Error Cantidades: 36.9%

In [171]:
eNoAnimal = pNoAnimal * (95.94/100) * (1.45/100)
print('Error de Imagenes sin Animales: ', round(eNoAnimal * 100,2), '%')

Error de Imagenes sin Animales:  0.95 %


In [172]:
eAnimal = pAnimal * (49.19/100) * ((1.41/100) + (41.79/100) * (5.22/100))
print('Error de Imagenes con Animales: ', round(eAnimal * 100,2), '%')

Error de Imagenes con Animales:  0.56 %


In [173]:
eAutomatico = ((3.8/100)+(5.1/100)+(36.9/100))*pAnimal+(2.49/100)*pNoAnimal
print('Error Automatico: ', round(eAutomatico * 100,2), '%')

Error Automatico:  16.3 %


In [174]:
cErroresAutom = tBase * eAutomatico
print('Cantidad de Errores: ', math.ceil(cErroresAutom))

Cantidad de Errores:  19357


### Costos del Proceso:

- Costo Falso Negativo: $300
- Costo Falso Positivo: $1000
- Costo Error en Especies: $1300
- Costo Error en Cantidad: $650

In [175]:
coErrorNoAnimal = eNoAnimal * 300
print('Costo de Error de Imagenes sin Animales: $', round(coErrorNoAnimal,2))

Costo de Error de Imagenes sin Animales: $ 2.84


In [176]:
coErrorAnimal = pAnimal * pGuanaco * ((1.41/100) * 1300 + (41.79/100) * (5.22/100) * 650)
print('Costo de Error de Imagenes con Animales: $', round(coErrorAnimal,2))

Costo de Error de Imagenes con Animales: $ 5.1


In [177]:
coErrorAutomatico = coErrorNoAnimal + coErrorAnimal
print('Costo de Error Automatico: $', round(coErrorAutomatico,2))

Costo de Error Automatico: $ 7.94


In [178]:
coOperacional = 0
print('Costo Operacional: $', round(coOperacional,2))

Costo Operacional: $ 0


In [179]:
coServicio = 1100000
print('Costo de Servicio: $', round(coServicio,2))

Costo de Servicio: $ 1100000


In [180]:
coTotal = tBase * (coErrorAutomatico + coOperacional) + coServicio
print('Costo Total: $', round(coTotal,2))

Costo Total: $ 2043139.28


In [181]:
coImagen = coTotal/tBase
print('Costo por Imagen: $', round(coImagen,2))

Costo por Imagen: $ 17.2


## KPIs

### Tiempo Total del Proceso

In [182]:
print('Tiempo Total: ')
print('- Clasificacion Manual: 4.63 segundos por imagen')
print('- Solucion Automatica:', round(tAutom,2), 'segundos por imagen')
print('Variacion Porcentual: ', round((tAutom - 4.63)/4.63 * 100,2), '%')

Tiempo Total: 
- Clasificacion Manual: 4.63 segundos por imagen
- Solucion Automatica: 0.56 segundos por imagen
Variacion Porcentual:  -88.0 %


In [183]:
print('Jornadas Laborales:')
print('- Clasificacion Manual: 19.1 jornadas')
print('- Solucion Automatica:')
print('    - Jornadas Totales: ', round(TAutom/8,1), 'jornadas')
print('    Variacion Porcentual: ', round((TAutom/8 - 19.1)/19.1 * 100,2), '%')

Jornadas Laborales:
- Clasificacion Manual: 19.1 jornadas
- Solucion Automatica:
    - Jornadas Totales:  2.3 jornadas
    Variacion Porcentual:  -88.0 %


### Cantidad de Imagenes Analizadas por Hora:

In [184]:
print('Cantidad de Imagenes por Hora: ')
print('- Clasificacion Manual: 777 imagenes/hr')
print('- Solucion Automatica: ', math.ceil(cHora), 'imagenes/hr')
print('Variacion Porcentual: ', round((cHora - 777)/777 * 100,2), '%')

Cantidad de Imagenes por Hora: 
- Clasificacion Manual: 777 imagenes/hr
- Solucion Automatica:  6480 imagenes/hr
Variacion Porcentual:  733.98 %


### Precision de Etiquetado:

In [185]:
print('Tasa de Error: ')
print('- Clasificacion Manual: 10%')
print('- Solucion Automatica: ', round(eAutomatico * 100,2), '%')
print('Variacion Porcentual: ', round((eAutomatico - 10)/10 * 100,2), '%')

Tasa de Error: 
- Clasificacion Manual: 10%
- Solucion Automatica:  16.3 %
Variacion Porcentual:  -98.37 %


In [186]:
print('Cantidad de Errores: ')
print('- Clasificacion Manual: 11877 errores')
print('- Solucion Automatica: ', math.ceil(cErroresAutom), 'errores')
print('Variacion Porcentual: ', round((cErroresAutom - 11877)/11877 * 100,2), '%')

Cantidad de Errores: 
- Clasificacion Manual: 11877 errores
- Solucion Automatica:  19357 errores
Variacion Porcentual:  62.97 %


### Costo del Proceso:

In [187]:
print('Costo por Imagen:')
print('- Clasificacion Manual: $ 56.5')
print('- Solucion Automatica: $', round(coImagen,2))
print('Variacion Porcentual: ', round((coImagen - 56.5)/56.5 * 100,2), '%')

Costo por Imagen:
- Clasificacion Manual: $ 56.5
- Solucion Automatica: $ 17.2
Variacion Porcentual:  -69.55 %


In [188]:
print('Costo del Proceso:')
print('- Clasificacion Manual: $ 6722325.4')
print('- Solucion Automatica: $', round(coTotal,2))
print('Variacion Porcentual: ', round((coTotal - 6722325.4)/6722325.4 * 100,2), '%')

Costo del Proceso:
- Clasificacion Manual: $ 6722325.4
- Solucion Automatica: $ 2043139.28
Variacion Porcentual:  -69.61 %


### Ratio de Imagenes Utiles:

In [189]:
print('Ratio de Imagenes Utiles: ')
print('- Clasificacion Manual: 0.79%')
print('- Solucion Automatica: 100%')
print('Variacion Porcentual: ', round((100 - 0.79)/0.79 * 100,2), '%')

Ratio de Imagenes Utiles: 
- Clasificacion Manual: 0.79%
- Solucion Automatica: 100%
Variacion Porcentual:  12558.23 %


In [190]:
EstimacionAutomatica = [round(tAutom,2), 0, round(TAutom/8,1), math.ceil(cHora), round(eAutomatico * 100,2), math.ceil(cErroresAutom), round(coImagen,2), 100]

# Solucion Semi-Automatica:

## Tiempo Total del Proceso:

In [191]:
tAutom = 0.1 + 0.07 + pAnimal * (0.07 + pGuanaco * 8)
TAutom = (tBase * (tAutom))/(60*60)
print('Tiempo de Automatizacion: ', round(tAutom,2), 'segundos por imagen')
print('Tiempo de Automatizacion: ', math.ceil(TAutom/8), 'jornadas')

Tiempo de Automatizacion:  1.45 segundos por imagen
Tiempo de Automatizacion:  6 jornadas


### Tiempo de Validacion:

El porcantaje de imagenes a validar sobre el total de la base es de:

In [192]:
# Porcentaje de Imagenes a Validar:
pVal = 0.2857
print('Porcentaje de Imagenes a Validar: ', round(pVal * 100,2), '%')

Porcentaje de Imagenes a Validar:  28.57 %


In [193]:
# Tiempo de Validacion:
tVal = pVal * 1.36
TVal = (tBase * tVal)/(60*60)
print('Tiempo de Validacion: ', round(TVal,2), 'horas')
print('Tiempo de Validacion: ', math.ceil(TVal/8), 'jornadas')

Tiempo de Validacion:  12.82 horas
Tiempo de Validacion:  2 jornadas


Por lo que el tiempo total del proceso de validacion es de **14.85 horas**  que son equivalentes a **2 jornadas**.

### Tiempo Total

El tiempo total del proceso sera de:

In [194]:
TTotal = TAutom + TVal
tTotal = tAutom + tVal
print('Tiempo del Proceso: ', round(TTotal,2), 'segundos por imagen')
print('Tiempo del Proceso: ', round(TTotal,2), 'horas')
print('Tiempo del Proceso: ', round(TTotal/8,2), 'jornadas')

Tiempo del Proceso:  60.55 segundos por imagen
Tiempo del Proceso:  60.55 horas
Tiempo del Proceso:  7.57 jornadas


El proceso completo requiere **2.84 jornadas laborales**  (23 horas). Cabe destacar que de estas solo **15 horas requieren atencion humana**

### Cantidad de Imagenes por Hora:

Para calcular la cantidad de imágenes que se procesaran por hora se tiene en cuenta tanto el tiempo que tarda en procesarlas los modelos predictivos cómo el tiempo que se tarda en validar aquellas que necesiten un refuerzo en su clasificación. 

In [195]:
print('Tiempo de Automatizacion: ', round(tAutom,2), 'segundos')

cHora = 60*60/(tAutom + tVal)
print('Cantidad de Imagenes por Hora: ', math.ceil(cHora))

Tiempo de Automatizacion:  1.45 segundos
Cantidad de Imagenes por Hora:  1962


### Precision de Etiquetado:

Se valida el 90% de las imagenes mal etiquetadas. La tasa de error en la validacion es del 5%.

In [196]:
eNoAnimal = pNoAnimal * (95.94/100) * (1.45/100)
print('Error de Imagenes sin Animales: ', round(eNoAnimal * 100,2), '%')

Error de Imagenes sin Animales:  0.95 %


In [197]:
eAnimal = pAnimal * (49.19/100) * ((1.41/100) + (41.79/100) * (5.22/100))
print('Error de Imagenes con Animales: ', round(eAnimal * 100,2), '%')

Error de Imagenes con Animales:  0.56 %


In [198]:
eAutomatico = eNoAnimal + eAnimal
print('Error Automatico: ', round(eAutomatico * 100,2), '%')

Error Automatico:  1.51 %


In [199]:
eErrores = 0.1 * eAutomatico + 0.9 * (5/100)
print('Error Total: ', round(eErrores * 100,2), '%')

Error Total:  4.65 %


In [200]:
cErrores = tBase * eErrores
print('Cantidad de Errores: ', math.ceil(cErrores))

Cantidad de Errores:  5525


### Costos del Proceso:

- Costo Falso Negativo: $300
- Costo Falso Positivo: $1000
- Costo Error en Especies: $1300
- Costo Error en Cantidad: $650

In [201]:
coErrorNoAnimal = eNoAnimal * 300
print('Costo de Error de Imagenes sin Animales: $', round(coErrorNoAnimal,2))

Costo de Error de Imagenes sin Animales: $ 2.84


In [202]:
coErrorAnimal = pAnimal * pGuanaco * ((1.41/100) * 1300 + (41.79/100) * (5.22/100) * 650)
print('Costo de Error de Imagenes con Animales: $', round(coErrorAnimal,2))

Costo de Error de Imagenes con Animales: $ 5.1


In [203]:
coErrorAutomatico = coErrorNoAnimal + coErrorAnimal
print('Costo de Error Automatico: $', round(coErrorAutomatico,2))

Costo de Error Automatico: $ 7.94


In [204]:
pValAnimal = 0.8094
pValNoAnimal = 0.406

In [205]:
coErrorVal = pAnimal * pValAnimal * ((1.41/100) * 300 + (41.79/100) * (5.22/100) * 650)/2 + pNoAnimal * pValNoAnimal * ((0.6/100)/2) * 1000
print('Costo de Error de Validacion: $', round(coErrorVal,2))

Costo de Error de Validacion: $ 3.2


In [206]:
coOperacionalVal = pVal * tVal * 10730/(8*60*60)
print('Costo Operacional de Validacion: $', round(coOperacionalVal,2))

Costo Operacional de Validacion: $ 0.04


In [207]:
coServicio = 1100000
print('Costo de Servicio: $', round(coServicio,2))

Costo de Servicio: $ 1100000


In [208]:
coTotal = tBase * (coErrorAutomatico + coErrorVal + coOperacionalVal) + coServicio
print('Costo Total: $', round(coTotal,2))

Costo Total: $ 2428687.68


In [209]:
coImagen = coTotal/tBase
print('Costo por Imagen: $', round(coImagen,2))

Costo por Imagen: $ 20.45


## KPIs

### Tiempo Total del Proceso

In [210]:
print('Tiempo Total: ')
print('- Clasificacion Manual: 4.63 segundos por imagen')
print('- Solucion Semi-Automatica:', round(tAutom,2), 'segundos por imagen')
print('Variacion Porcentual: ', round((tAutom - 4.63)/4.63 * 100,2), '%')

Tiempo Total: 
- Clasificacion Manual: 4.63 segundos por imagen
- Solucion Semi-Automatica: 1.45 segundos por imagen
Variacion Porcentual:  -68.75 %


In [211]:
print('Jornadas Laborales:')
print('- Clasificacion Manual: 19.1 jornadas')
print('- Solucion Semi-Automatica:')
print('    - Jornadas Totales: ', round(TAutom/8,1), 'jornadas')
print('    Variacion Porcentual: ', round((TAutom/8 - 19.1)/19.1 * 100,2), '%')

Jornadas Laborales:
- Clasificacion Manual: 19.1 jornadas
- Solucion Semi-Automatica:
    - Jornadas Totales:  6.0 jornadas
    Variacion Porcentual:  -68.76 %


### Cantidad de Imagenes Analizadas por Hora:

In [212]:
print('Cantidad de Imagenes por Hora: ')
print('- Clasificacion Manual: 777 imagenes/hr')
print('- Solucion Semi-Automatica: ', math.ceil(cHora), 'imagenes/hr')
print('Variacion Porcentual: ', round((cHora - 777)/777 * 100,2), '%')

Cantidad de Imagenes por Hora: 
- Clasificacion Manual: 777 imagenes/hr
- Solucion Semi-Automatica:  1962 imagenes/hr
Variacion Porcentual:  152.43 %


### Precision de Etiquetado:

In [213]:
print('Tasa de Error: ')
print('- Clasificacion Manual: 10%')
print('- Solucion Semi-Automatica: ', round(eAutomatico * 100,2), '%')
print('Variacion Porcentual: ', round((eAutomatico - 10)/10 * 100,2), '%')

Tasa de Error: 
- Clasificacion Manual: 10%
- Solucion Semi-Automatica:  1.51 %
Variacion Porcentual:  -99.85 %


In [214]:
print('Cantidad de Errores: ')
print('- Clasificacion Manual: 11877 errores')
print('- Solucion Semi-Automatica: ', math.ceil(cErroresAutom), 'errores')
print('Variacion Porcentual: ', round((cErroresAutom - 11877)/11877 * 100,2), '%')

Cantidad de Errores: 
- Clasificacion Manual: 11877 errores
- Solucion Semi-Automatica:  19357 errores
Variacion Porcentual:  62.97 %


### Costo del Proceso:

In [215]:
print('Costo por Imagen:')
print('- Clasificacion Manual: $ 56.5')
print('- Solucion Semi-Automatica: $', round(coImagen,2))
print('Variacion Porcentual: ', round((coImagen - 56.5)/56.5 * 100,2), '%')

Costo por Imagen:
- Clasificacion Manual: $ 56.5
- Solucion Semi-Automatica: $ 20.45
Variacion Porcentual:  -63.81 %


In [216]:
print('Costo del Proceso:')
print('- Clasificacion Manual: $ 6722325.4')
print('- Solucion Semi-Automatica: $', round(coTotal,2))
print('Variacion Porcentual: ', round((coTotal - 6722325.4)/6722325.4 * 100,2), '%')

Costo del Proceso:
- Clasificacion Manual: $ 6722325.4
- Solucion Semi-Automatica: $ 2428687.68
Variacion Porcentual:  -63.87 %


### Ratio de Imagenes Utiles:

In [217]:
print('Ratio de Imagenes Utiles: ')
print('- Clasificacion Manual: 0.79%')
print('- Solucion Semi-Automatica: 100%')
print('Variacion Porcentual: ', round((100 - 0.79)/0.79 * 100,2), '%')

Ratio de Imagenes Utiles: 
- Clasificacion Manual: 0.79%
- Solucion Semi-Automatica: 100%
Variacion Porcentual:  12558.23 %


In [218]:
EstimacionSemiAutomatica = [round(tTotal,2), round(TVal/8,1), round(TTotal/8,1), math.ceil(cHora), round(eErrores * 100,2), math.ceil(cErrores), round(coImagen,2), 100]

# Resultados:

In [219]:
# Creación del DataFrame
data = {
    "KPI": [
        "Segundos por Imagen",
        "Jornadas Ocupadas",
        "Jornadas Totales",
        "Imagenes por Hora",
        "Tasa de Error",
        "Cantidad de Errores",
        "Costo por Imagen",
        "Ratio de Imagenes Utiles"
    ],
    "Clasificacion Manual": [
        4.63, # Tiempo Total
        19.1, # Jornadas Ocupadas
        19.1, # Jornadas Totales
        777, # Imagenes por Hora
        10, # Tasa de Error
        11877, # Cantidad de Errores
        56.5, # Costo del Proceso
        0.79 # Ratio de Imagenes Utiles
    ],
    "Estimacion Automatica": EstimacionAutomatica,
    "Estimacion Semi-Automatica": EstimacionSemiAutomatica,
}

df = pd.DataFrame(data)
# df["Variacion Porcentual (%)"] = round((df["Estimacion Semi-Automatica"] - df["Clasificacion Manual"]) / df["Clasificacion Manual"] * 100, 2)
df

KPI  Clasificacion Manual  Estimacion Automatica  \
0       Segundos por Imagen                  4.63                   0.56   
1         Jornadas Ocupadas                 19.10                   0.00   
2          Jornadas Totales                 19.10                   2.30   
3         Imagenes por Hora                777.00                6480.00   
4             Tasa de Error                 10.00                  16.30   
5       Cantidad de Errores              11877.00               19357.00   
6          Costo por Imagen                 56.50                  17.20   
7  Ratio de Imagenes Utiles                  0.79                 100.00   

   Estimacion Semi-Automatica  
0                        1.84  
1                        1.60  
2                        7.60  
3                     1962.00  
4                        4.65  
5                     5525.00  
6                       20.45  
7                      100.00

In [220]:
df.to_csv('resultados.csv', index=False)